integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/ingest/CreateElasticsearchBackedHiveTable.java

```diff
@@ -0,0 +1,275 @@
+        .build();
+
+    public static final PropertyDescriptor USE_WAN = new PropertyDescriptor.Builder()
+        .name("Use WAN")
+        .description(
+            "Whether the connector is used against an Elasticsearch instance in a cloud/restricted environment over the WAN, such as Amazon Web Services. "
+            + "In this mode, the connector disables discovery and only connects through the declared es.nodes during all operations, including reads and writes."
+            + " Note that in this mode, performance is highly affected.")
+        .allowableValues("true", "false")
+        .required(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(false)
+        .defaultValue("true")
+        .build();
+
+    public static final PropertyDescriptor AUTO_CREATE_INDEX = new PropertyDescriptor.Builder()
+        .name("Auto create index")
+        .description("Whether or not the Elasticsearch index can be created if id does not already exist.")
+        .allowableValues("true", "false")
+        .required(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(false)
+        .defaultValue("true")
+        .build();
+
+    public static final PropertyDescriptor FIELD_INDEX_STRING = new PropertyDescriptor.Builder()
+        .name("Field Index String")
+        .description("Comma separated list of fields to be indexed.")
+        .required(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(true)
+        .defaultValue("${metadata.table.fieldIndexString}")
+        .build();
+
+
+    private final Set<Relationship> relationships;
+    private final List<PropertyDescriptor> propDescriptors;
+
+    /**
+     * default constructor constructs the relationship and property collections
+     */
+    public CreateElasticsearchBackedHiveTable() {
+        final Set<Relationship> r = new HashSet<>();
+        r.add(IngestProperties.REL_SUCCESS);
+        r.add(IngestProperties.REL_FAILURE);
+        relationships = Collections.unmodifiableSet(r);
+
+        final List<PropertyDescriptor> pds = new ArrayList<>();
+        pds.add(NODES);
+        pds.add(ID_FIELD);
+        pds.add(FIELD_SPECIFICATION);
+        pds.add(FEED_NAME);
+        pds.add(FEED_CATEGORY);
+        pds.add(THRIFT_SERVICE);
+        pds.add(JAR_URL);
+        pds.add(USE_WAN);
+        pds.add(AUTO_CREATE_INDEX);
+        pds.add(FIELD_INDEX_STRING);
+        propDescriptors = Collections.unmodifiableList(pds);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propDescriptors;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.apache.nifi.processor.exception.ProcessException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)